suite/suite.go

```diff
@@ -193,12 +232,13 @@ func runTests(t testing.TB, tests []testing.InternalTest) {
 
 	r, ok := t.(runner)
 	if !ok { // backwards compatibility with Go 1.6 and below
+		// t.F doesn't have Run() so Fuzz tests also
+		// take this branch
 		if !testing.RunTests(allTestsFilter, tests) {
 			t.Fail()
```

Comment:
What i am observing is this which explains the behavior I am seeing (which may have been clear to you folks, but i am seeing why now):
- When the fuzz test is run in "unit test" mode, the function call `RunTests()` returns with a `false` value and hence the test fails (considering the sample test in the PR)
- When the fuzz test is run in fuzz mode, this function returns `true` 
-

Reply:
I am going to next experiment with updating the test discovery and running logic using https://pkg.go.dev/testing@go1.18beta2#InternalFuzzTarget  for the fuzz tests. i think that may be required and certainly feels correct.

Reply:
> Okay, I think i have found the issue. The problem seems to be that when we call the `suite.EqualError()` method, it reports any failures using the testing context's (`testing.T` value) methods. However, the problem is since we are running a Fuzz test, we only have a `testing.F` injected via `go test`. The `testing.T` value is created and injected into the fuzz target [as part of the test run](https://sourcegraph.com/github.com/golang/go@38174b3a3514629b84dcd76878b2f536b189dd7b/-/blob/src/testing/fuzz.go?L302). So, we need to make this `testing.T` created as part of the test run available to testify.

Hence, when i update my test as follows to make a call to the `SetT()` method, it fails as expected in fuzz mode as well:

```
type ExampleFuzzSuite struct {
	suite.Suite
}

func (suite *ExampleFuzzSuite) FuzzAFunc() {
	f := suite.F()
	f.Add("foo")
	f.Fuzz(func(t *testing.T, aStr string) {
		fmt.Println("I am in Fuzz target")
		// this fails the test run when run in fuzz mode
		//t.Fail()

		// this doesn't fail the test when run in fuzz mode if i comment
		// out the next line: https://github.com/stretchr/testify/pull/1142/files#r821269235
		suite.SetT(t)
		suite.EqualError(nil, "an error")
	})
}

func TestExampleTestSuite(t *testing.T) {
	suite.Run(t, new(ExampleTestSuite))
}

func FuzzSuite(f *testing.F) {
	suite.Run(f, new(ExampleFuzzSuite))
}
```